### PR TITLE
pin `asdf` to newest patch version to fix dependency for newest `jsonschema` release

### PR DESCRIPTION
**Pull request by zacharyburnett**
related to https://github.com/asdf-format/asdf/pull/1171; when `asdf 2.12.1` is released, this PR can be merged

**Checklist for maintainers**
- [x] added entry in `CHANGES.rst` within the relevant release section
- [ ] updated or added relevant tests
- [ ] updated relevant documentation
- [x] added relevant milestone
- [x] added relevant label(s)
- [ ] ran regression tests, post a link to the Jenkins job below.
      [How to run regression tests on a PR](https://github.com/spacetelescope/jwst/wiki/Running-Regression-Tests-Against-PR-Branches)
- [ ] Make sure the JIRA ticket is [resolved properly](https://github.com/spacetelescope/jwst/wiki/How-to-resolve-JIRA-issues)

### CODE REVIEW

**Review by WilliamJamieson — APPROVED** [review]
Looks good to me.

**Comment by WilliamJamieson** [comment]
closing and reopening

File: CHANGES.rst [hunk 1]
```diff
@@ -6,7 +6,7 @@ general
 
 - Made code style changes due to the new 5.0.3 version of flake8, which noted many
   missing white spaces after keywords. [#6958]
-- pin `jsonschema` below `4.10.0` to fix issues with unit and regression tests [#6986]
+- pin ``asdf`` above ``2.12.1`` to fix issues with unit and regression tests [#6986, #6991]
```
[the remark this answers is not in the document]
Comment (hbushouse) [reply CHANGES.rst:9]: OK, so in that case I would not characterize this fix in the change log as merely being necessary to fix the tests, but rather to fix fundamental problems with datamodel I/O for the entire pipeline!  So perhaps something like "to fix issues encountered within ASDF due to jsonschema 4.10.x" (or better wording that you can come up with).

**Review by hbushouse — APPROVED** [review]

**Comment by hbushouse** [comment]
Looks like the couple of CI failures are the usual random hiccups trying to get ref files from CRDS. Merging.

**Pull request merged**